usr/sbin/p11sak/p11sak.c

```diff
@@ -2537,7 +2616,44 @@ static CK_RV list_ckey(CK_SESSION_HANDLE session, p11sak_kt kt, int long_print)
             printf("Label: %s\t\t", label);
         }
 
+        uri = p11_uri_new();
+        if (!uri)
+            return CKR_HOST_MEMORY;
+
+        uri->info = &info;
+        uri->slot_id = slot;
+        uri->slot_info = &slot_info;
+        uri->token_info = &token_info;
+
+        if (CKR_OK == tok_attribute_alloc(session, hkey, uri->obj_class)) {
```

Comment:
we usually have it the other way around, i.e. `tok_attribute_alloc(session, hkey, uri->obj_class) == CKR_OK`.

Reply:
ok